server/src/main/java/com/cloud/api/ApiResponseHelper.java

```diff
@@ -2598,6 +2598,9 @@ public NetworkResponse createNetworkResponse(ResponseView view, Network network)
             if (domain != null) {
                 response.setDomainId(domain.getUuid());
                 response.setDomainName(domain.getName());
+                StringBuilder domainPath = new StringBuilder("ROOT");
```

Comment:
@isupersky , I think we should not prepend "ROOT" in case there is any other path following (like "ROOT/mydomain").
We could replace if it is "/", I have no preference on that.

Reply:
@DaanHoogland  Please correct me if I'm wrong

what I understand is -
- "ROOT" should not be prepended  in any domainPath added in this PR.
- and if somewhere this has been done previously also then it should be replaced with "/"

Reply:
no, that is not what I meant
if the root domain is the it we return "/" or instead "ROOT", or even "".
in all other cases we return something like "/domain/subdomain/..."

I would not bother about other places withing the scope of this PR (unless you feel like it)

Reply:
looks like you went for the empty string right?

Reply:
yes, or else we can go with ->

`StringBuilder domainPath = new StringBuilder(); `
        `if(ObjectUtils.isEmpty(domain.getPath())){`
           ` domainPath.append("/");`
       ` }`
        `else{`
           ` (domainPath.append(domain.getPath())).deleteCharAt(domainPath.length() - 1);`
       ` }`
        `response.setDomainPath(domainPath.toString());`

Reply:
i like that, to creep the scope a bit, we could even make the "/" a configuration item ;)
but lgtm!!!

Reply:
made the changes.  

Not sure, what you mean by configurable here.  is it that rather than using a string "/" multiple times, i should extract it in single variable? 

something like -> `private static final String defaultDomainPath = "/";`

Reply:
You can have a look at the uses of `ConfigKey<>` , you'd be using something like `defaultRootDomainName.value()` and the operator can than change that. But let's not do that. It is fine like so.